temporian/implementation/numpy_cc/operators/custom_heap.h

```diff
@@ -0,0 +1,76 @@
+#include <iostream>
+#include <list>
+template <typename T>
+class CustomHeap {
+ private:
+  std::function<bool(T, T)> compare;
```

Comment:
std::functions can be surprisingly slow. Can the compactor be a template argument?